x-pack/plugins/canvas/server/lib/get_es_filter.ts

```diff
@@ -10,16 +10,27 @@
   filter is the abstracted canvas filter.
 */
 
-/*eslint import/namespace: ['error', { allowComputed: true }]*/
+import {
+  CanvasQueryFilter,
+  ElasticsarchTermFilter,
+  ElasticsearchLuceneQueryStringFilter,
+  ElasticsearchTimeFilter,
+} from './filters';
 import * as filters from './filters';
 
-export function getESFilter(filter) {
-  if (!filters[filter.type]) {
+interface FilterTypes {
+  [key: string]: any;
```

Comment:
Aren't these all functions and consts?  Perhaps you should export a typed collection for import elsewhere... because you're having to cast it strangely below.